tests/integration/src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala

```diff
@@ -760,6 +759,39 @@ class AppDeployIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathon
       waitForTasks(id, 2) //make sure, all the tasks have really started
     }
 
+    // Regression for MARATHON-8537
+    "rollback deployment with default deployment strategy" in {
+      Given("an existing app")
+      val id: PathId = appId(Some("yet-another-deployment-rollback"))
+      val app = App(
+        id = id.toString,
+        cmd = Some("sleep 12345"),
+        backoffFactor = 1d
+      )
```

Comment:
Hm, how is this App definition different from `rollback a deployment` https://github.com/mesosphere/marathon/blob/master/tests/integration/src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala#L693? Both do not seem to set the `UpgradeStrategy`.

Reply:
that test is rolling back FIRST deployment so it's basically removing the app. That's why I removed the test in this PR as well, kind of misleading... I did not even know that can happen :D that removing an app is valid rollback :)

Reply:
Oh, the first test should be renamed to remove app in deployment or something.